### PR TITLE
Add handler unit tests with mock clients

### LINKED ISSUE #76: Part 2: Add handler unit tests with mock clients
Part of #74 — depends on Part 1

Add function-field mock structs and write tests exercising the real handler methods (ServeConditionalBoot, ServeAnswer, ServeISOContent) with httptest.

**Files:** `internal/handlers/boot_test.go`, `answer_test.go`, `iso_test.go`

### PR DESCRIPTION
**Pull request by twdamhore**
## Summary

- Add `mockBootClient`, `mockAnswerClient`, `mockISOClient` structs using function fields
- Add tests for real handler methods: `ServeBootIPXE`, `ServeConditionalBoot`, `ServeBootDone`, `ServeAnswer`, `ServeISOContent`
- Cover success paths, error paths (gRPC errors, not found), and edge cases (missing MAC, no pending provision, invalid disk image ref)

Part of #74. Closes #76. Depends on #75 (merged).

## Test plan

- [x] `go test ./internal/handlers/ -v` — all 49 tests pass
- [x] `go test ./...` — full suite passes

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: internal/handlers/boot_test.go [hunk 1]
```diff
@@ -196,3 +224,191 @@ func TestSplitHostDomain(t *testing.T) {
...
+	h := NewBootHandler("10.0.0.1", "8080", "3128", mock, "cm")
+	req := httptest.NewRequest("GET", "/boot/done?mac=aa-bb-cc-dd-ee-ff", nil)
+	w := httptest.NewRecorder()
+
+	h.ServeBootDone(w, req)
+
+	if w.Code != http.StatusNotFound {
+		t.Fatalf("expected 404, got %d", w.Code)
+	}
+}
```
[the remark this answers is not in the document]
Comment (twdamhore) [reply internal/handlers/boot_test.go]: Added `TestServeConditionalBoot_NoMAC_Real` and `TestServeBootDone_NoMAC_Real` — both call the real handler methods with no `mac` query param and verify 400 Bad Request.

[the remark this answers is not in the document]
Comment (twdamhore) [reply internal/handlers/boot_test.go]: Covered by same tests added for the first thread: `TestServeConditionalBoot_NoMAC_Real` and `TestServeBootDone_NoMAC_Real`.

File: internal/handlers/boot_test.go [hunk 2]
```diff
@@ -196,3 +224,272 @@ func TestSplitHostDomain(t *testing.T) {
...
+	}
+	if w.Body.String() != "ok" {
+		t.Errorf("expected body 'ok', got %q", w.Body.String())
+	}
+	if updatedName != "prov-1" || updatedStatus != "Complete" {
+		t.Errorf("expected prov-1 marked Complete, got name=%q status=%q", updatedName, updatedStatus)
+	}
+}
+
+func TestServeConditionalBoot_NoMAC_Real(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (twdamhore) [reply internal/handlers/boot_test.go]: Renamed to `TestServeConditionalBoot_NoMAC` — removed the old inline test that had the same name and dropped the `_Real` suffix.

File: internal/handlers/boot_test.go [hunk 3]
```diff
@@ -196,3 +224,272 @@ func TestSplitHostDomain(t *testing.T) {
 		}
 	}
 }
+
+func TestServeBootIPXE_Real(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (twdamhore) [reply internal/handlers/boot_test.go]: Renamed to `TestServeBootIPXE_Success` for consistency with the `_Success`/`_Error` pattern used by other handler tests.
